core/ops.rs

```diff
@@ -72,30 +73,38 @@ pub struct OpCtx {
   #[doc(hidden)]
   pub get_error_class_fn: GetErrorClassFn,
 
-  pub(crate) decl: Rc<OpDecl>,
+  #[doc(hidden)]
+  pub decl: Rc<OpDecl>,
   pub(crate) fast_fn_c_info: Option<NonNull<v8::fast_api::CFunctionInfo>>,
-  pub(crate) metrics_fn: Option<OpMetricsFn>,
+  #[doc(hidden)]
+  pub metrics_fn: Option<OpMetricsFn>,
   /// If the last fast op failed, stores the error to be picked up by the slow op.
   pub(crate) last_fast_error: UnsafeCell<Option<AnyError>>,
 
-  context_state: Rc<ContextState>,
+  op_driver: Rc<DefaultOpDriver>,
   runtime_state: Rc<JsRuntimeState>,
 }
 
 impl OpCtx {
   #[allow(clippy::too_many_arguments)]
   pub(crate) fn new(
+    // TODO(bartlomieju): remove
+    op_count: usize,
```

Comment:
I really don't like this, but `OpMetricsFactoryFn` requires it. It appears we only use the total count in `OpMetricsSummaryTracker` that is only used in testing code.

Could we maybe remove `OpMetricsFactoryFn` to accept the total count of ops?

Reply:
If that's okay, then maybe this factory function is not needed anymore?